packages/react/src/blocks/ToggleWrapper/ToggleWrapper.tsx

```diff
@@ -0,0 +1,97 @@
+import { Block } from "@blocknote/core";
+import { ReactNode, useEffect, useMemo, useState } from "react";
+
+import { ReactCustomBlockRenderProps } from "../../schema/ReactBlockSpec.js";
+import { useEditorChange } from "../../hooks/useEditorChange.js";
+
+export const ToggleWrapper = (
+  props: Omit<ReactCustomBlockRenderProps<any, any, any>, "contentRef"> & {
+    children: ReactNode;
+    toggledState?: {
+      set: (block: Block<any, any, any>, isToggled: boolean) => void;
+      get: (block: Block<any, any, any>) => boolean;
+    };
+  },
+) => {
+  const { block, editor, children } = props;
+
+  const toggledState = useMemo(
+    () =>
+      props.toggledState || {
+        set: (block, isToggled: boolean) =>
+          window.localStorage.setItem(`toggle-${block.id}`, String(isToggled)),
+        get: (block) =>
+          window.localStorage.getItem(`toggle-${block.id}`) === "true",
+      },
```

Comment:
Is constant, should be extracted to module scope
can remove memo too, since it will always be broken or valid